src/Symfony/Component/Form/Extension/Core/Type/CollectionType.php

```diff
@@ -74,23 +74,45 @@ public function finishView(FormView $view, FormInterface $form, array $options)
      */
     public function configureOptions(OptionsResolver $resolver)
     {
-        $optionsNormalizer = function (Options $options, $value) {
+        $entryOptionsNormalizer = function (Options $options, $value) {
             $value['block_name'] = 'entry';
 
             return $value;
         };
+        $entryType = function (Options $options) {
+            if (null !== $options['type']) {
+                trigger_error('The form option "type" is deprecated since version 2.7 and will be removed in 3.0. Use "entry_type" instead.', E_USER_DEPRECATED);
+
+                return $options['type'];
+            }
+
+            return 'text';
+        };
+        $entryOptions = function (Options $options) {
+            if (null !== $options['options']) {
+                trigger_error('The form option "options" is deprecated since version 2.7 and will be removed in 3.0. Use "entry_options" instead.', E_USER_DEPRECATED);
+
+                return $options['options'];
+            }
+
+            return array();
+        };
 
         $resolver->setDefaults(array(
             'allow_add' => false,
             'allow_delete' => false,
             'prototype' => true,
             'prototype_name' => '__name__',
-            'type' => 'text',
-            'options' => array(),
+            // deprecated as of Symfony 2.7, to be removed in Symfony 3.0. Use entry_type instead
+            'type' => null,
+            // deprecated as of Symfony 2.7, to be removed in Symfony 3.0. Use entry_options instead
+            'options' => null,
+            'entry_type' => $entryType,
+            'entry_options' => $entryOptions,
             'delete_empty' => false,
         ));
 
-        $resolver->setNormalizer('options', $optionsNormalizer);
+        $resolver->setNormalizer('entry_options', $entryOptionsNormalizer);
```

Comment:
Removing the normalizer from "options" breaks BC for anyone that currently accesses the "block_name" option (which disappears if removing the normalizer). I think you should keep the normalizer for both options.